@xen-orchestra/fs/src/utils.js

```diff
@@ -0,0 +1,45 @@
+import { createLogger } from '@xen-orchestra/log'
+import { TimeoutError } from 'promise-toolbox'
+const { info, warn } = createLogger('xo:fs:abstract')
+
+export function withTimeout(fn, timeout, { onSuccess, onTimeout, onSuccessAfterTimeout, onFailureAfterTimeout } = {}) {
+  if (typeof fn.then !== 'undefined') {
+    throw new Error('Function needs to be asynchronous.')
+  }
+
+  return function (...args) {
+    let timeoutHandle
+    let didTimeout = false
+    const timeoutError = new TimeoutError('Async call timeout limit reached')
```

Comment:
is the erro stack trace complete when it throws ? We need to at least know the caller of the base function